summingbird-batch/src/main/scala/com/twitter/summingbird/batch/BatchID.scala

```diff
@@ -57,8 +69,18 @@ object BatchID {
    * `[startBatch, endBatch]` (inclusive).
    */
   def range(start: BatchID, end: BatchID): Iterable[BatchID] =
-    new Iterable[BatchID] {
-      def iterator = iterate(start)(_.next).takeWhile(_ <= end)
+    if (start == end) List(start)
+    else if (end < start) Nil
+    else {
+      new Iterable[BatchID] {
```

Comment:
I'll do @pankajroark 's approach.

I like keeping this knowledge in the Successible instance which is designed to exactly handle this task.

I don't like that `.next` may wrap around and think that API may have been a mistake at this point, and Successible does not have that problem.